File: Can Place Flowers.go
package leet

func canPlaceFlowers(flowerbed []int, n int) bool {
	length := len(flowerbed)

	for i := 0; i < length; i++ {
		if flowerbed[i] == 0 {
			leftEmpty := (i == 0) || (flowerbed[i-1] == 0)
			rightEmpty := (i == length-1) || (flowerbed[i+1] == 0)

			if leftEmpty && rightEmpty {
				flowerbed[i] = 1
				n--

				if n == 0 {
					return true
				}
			}
		}
	}
	return n <= 0
}

File: Can place flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        length = len(flowerbed)

        for i in range(length):
            left = (i == 0) or (flowerbed[i-1] == 0)
            right = (i == length-1) or (flowerbed[i+1] == 0)

            if flowerbed[i] == 0:
                if left and right:
                    flowerbed[i] = 1
                    n -= 1
            
            if n == 0:
                return True
        return n <= 0